packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -2456,6 +2420,202 @@ contract OPContractsManager is ISemver {
         return opcmDeployer.isDevFeatureEnabled(_feature);
     }
 
+    /// @notice Helper that converts the legacy DeployInput into the new FullConfig.
+    /// @param _input The legacy DeployInput.
+    /// @param _superchainConfig The SuperchainConfig contract.
+    /// @return The new FullConfig.
+    function _toFullConfig(
+        DeployInput memory _input,
+        ISuperchainConfig _superchainConfig
+    )
+        internal
+        pure
+        returns (IOPContractsManagerV2.FullConfig memory)
+    {
+        // Start building the full config.
+        IOPContractsManagerV2.FullConfig memory cfg;
+
+        // Handle salt mixer.
+        cfg.saltMixer = _input.saltMixer;
+
+        // Handle system roles.
+        cfg.proxyAdminOwner = _input.roles.opChainProxyAdminOwner;
+        cfg.systemConfigOwner = _input.roles.systemConfigOwner;
+        cfg.unsafeBlockSigner = _input.roles.unsafeBlockSigner;
+        cfg.batcher = _input.roles.batcher;
+
+        // Handle L2 system configuration.
+        cfg.basefeeScalar = _input.basefeeScalar;
+        cfg.blobBasefeeScalar = _input.blobBasefeeScalar;
+        cfg.gasLimit = _input.gasLimit;
+        cfg.l2ChainId = _input.l2ChainId;
+        cfg.resourceConfig = Constants.DEFAULT_RESOURCE_CONFIG();
```

Comment:
hmmm, why is `resourceConfig` becoming a configurable input? Is that wanted or needed?

Reply:
This one is mostly just because it's hard to figure out how to pass this around if you don't have it as a field on `FullConfig`. It needs to make its way to the `_execute` function somehow. `resourceConfig` is already configurable by chain operators if they want it to be, easy enough to just set it to the recommended default value manually.